### PR TITLE
 Miri terminator handling: only do progress sanity check for 'Call' terminator

### LINKED ISSUE #70723: ICE with `#![feature(const_loop)] static _X: () = loop {};`
### Code

```Rust
#![feature(const_loop)]

static _X: () = loop {};
```

Also reproduces with `const` or `static mut` (add those in a regression test!).

### Meta

note: rustc 1.44.0-nightly (76b11980a 2020-04-01) running on x86_64-unknown-linux-gnu

### Error output

```
   Compiling playground v0.0.1 (/playground)
thread 'rustc' panicked at 'assertion failed: self.cur_frame() != old_stack || self.frame().block != old_bb', src/librustc_mir/interpret/step.rs:288:13
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1069
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1439
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:198
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:218
  10: rustc_driver::report_ice
  11: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:515
  12: std::panicking::begin_panic
  13: rustc_mir::interpret::step::<impl rustc_mir::interpret::eval_context::InterpCx<M>>::run
  14: rustc_mir::const_eval::eval_queries::const_eval_raw_provider
  15: rustc_middle::ty::query::<impl rustc_query_system::query::config::QueryAccessors<rustc_middle::ty::context::TyCtxt> for rustc_middle::ty::query::queries::const_eval_raw>::compute
  16: rustc_query_system::dep_graph::graph::DepGraph<K>::with_task_impl
  17: rustc_query_system::query::plumbing::get_query
  18: rustc_mir::const_eval::eval_queries::const_eval_validated_provider
  19: rustc_middle::ty::query::<impl rustc_query_system::query::config::QueryAccessors<rustc_middle::ty::context::TyCtxt> for rustc_middle::ty::query::queries::const_eval_validated>::compute
  20: rustc_query_system::dep_graph::graph::DepGraph<K>::with_task_impl
  21: rustc_query_system::query::plumbing::get_query
  22: rustc_mir::const_eval::eval_queries::const_eval_validated_provider
  23: rustc_middle::ty::query::<impl rustc_query_system::query::config::QueryAccessors<rustc_middle::ty::context::TyCtxt> for rustc_middle::ty::query::queries::const_eval_validated>::compute
  24: rustc_query_system::dep_graph::graph::DepGraph<K>::with_task_impl
  25: rustc_query_system::query::plumbing::get_query
  26: rustc_middle::mir::interpret::queries::<impl rustc_middle::ty::context::TyCtxt>::const_eval_global_id
  27: rustc_middle::mir::interpret::queries::<impl rustc_middle::ty::context::TyCtxt>::const_eval_poly
  28: <rustc_lint::BuiltinCombinedLateLintPass as rustc_lint::passes::LateLintPass>::check_item
  29: rustc_hir::intravisit::Visitor::visit_nested_item
  30: rustc_hir::intravisit::walk_crate
  31: rustc_lint::late::late_lint_crate
  32: rustc_data_structures::sync::join
  33: rustc_interface::passes::analysis::{{closure}}::{{closure}}
  34: rustc_session::utils::<impl rustc_session::session::Session>::time
  35: rustc_interface::passes::analysis
  36: rustc_middle::ty::query::<impl rustc_query_system::query::config::QueryAccessors<rustc_middle::ty::context::TyCtxt> for rustc_middle::ty::query::queries::analysis>::compute
  37: rustc_query_system::dep_graph::graph::DepGraph<K>::with_task_impl
  38: rustc_query_system::query::plumbing::get_query
  39: rustc_middle::ty::context::tls::enter_global
  40: rustc_interface::interface::run_compiler_in_existing_thread_pool
  41: scoped_tls::ScopedKey<T>::set
  42: rustc_ast::attr::with_globals
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.44.0-nightly (76b11980a 2020-04-01) running on x86_64-unknown-linux-gnu

note: compiler flags: -C codegen-units=1 -C debuginfo=2 --crate-type lib

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
#0 [const_eval_raw] const-evaluating `_X`
#1 [const_eval_validated] const-evaluating + checking `_X`
#2 [const_eval_validated] const-evaluating + checking `_X`
#3 [analysis] running analysis passes on this crate
end of query stack
error: could not compile `playground`.
```

### PR DESCRIPTION
**Pull request by RalfJung**
This will still catch mistakes in bad intrinsic/foreign-item shims, which is the main source of errors here.

Fixes https://github.com/rust-lang/rust/issues/70723
r? @oli-obk

### CODE REVIEW

File: src/test/ui/consts/const-eval/issue-70723.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#![feature(const_loop)]
+
+static _X: () = loop {}; //~ ERROR could not evaluate static initializer
```
Comment (oli-obk) [inline src/test/ui/consts/const-eval/issue-70723.rs:3]: how long does this test take? If more than a second, please lower the step limit

**Comment by oli-obk** [comment]
r=me after ensuring that the new test doesn't consume too much time

File: src/test/ui/consts/const-eval/issue-70723.rs (hunk 1 above)
Comment (RalfJung) [reply src/test/ui/consts/const-eval/issue-70723.rs:3]: It takes not even 200ms here:
```
$ time rustc +rustc.2-s1 src/test/ui/consts/const-eval/issue-70723.rs 
error[E0080]: could not evaluate static initializer
 --> src/test/ui/consts/const-eval/issue-70723.rs:3:17
  |
3 | static _X: () = loop {}; //~ ERROR could not evaluate static initializer
  |                 ^^^^^^^ exceeded interpreter step limit (see `#[const_eval_limit]`)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0080`.

real    0m0,125s
user    0m0,064s
sys     0m0,029s
```

**Pull request merged**